paramiko/rsakey.py

```diff
@@ -26,26 +26,31 @@
 from cryptography.hazmat.primitives.asymmetric import rsa, padding
 
 from paramiko.message import Message
-from paramiko.pkey import PKey
+from paramiko.pkey import PKey, register_pkey_type
 from paramiko.py3compat import PY2
 from paramiko.ssh_exception import SSHException
 
 
+@register_pkey_type
 class RSAKey(PKey):
     """
     Representation of an RSA key which can be used to sign and verify SSH2
     data.
     """
 
+    LEGACY_TYPE = 'RSA'
+    OPENSSH_TYPE_PREFIX = 'ssh-rsa'
```

Comment:
The `@register_pkey_type` decorator reads these attributes and stores them in a global, which keeps track of all known key types.

What's slightly iffy is that if a key class is never imported, then `load_private_key` cannot know about it. :(

But this should not be a problem, since the main `paramiko/__init__.py` contains these imports.